### PR TITLE
Add option to persist labels to the URL search query

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

**Comment by jwforres** [comment]
@spadgett

File: labelFilter.js [hunk 1]
```diff
@@ -34,6 +34,33 @@ angular.module('kubernetesUI')
     }
   };
 
+  LabelFilter.prototype.persistFilterState = function(persist) {
+    this._shouldPersistState = !!persist;
+  };
+
+  LabelFilter.prototype._persistState = function() {
+    if (this._shouldPersistState) {
```
Comment (spadgett) [inline labelFilter.js]: A little easier to read as

``` js
if (!this._shouldPersistState) {
  return;
}

...
```

File: labelFilter.js [hunk 2]
```diff
@@ -34,6 +34,33 @@ angular.module('kubernetesUI')
...
+      if (this._labelSelector.isEmpty()) {
+        $location.replace().search({labelFilter: null});
```
Comment (spadgett) [inline labelFilter.js]: Will this remove other query parameters from the URL?

File: labelSelector.js [hunk 3]
```diff
@@ -164,6 +164,29 @@ LabelSelector.prototype.covers = function(selector) {
...
 };
 
+// Exports the labelSelector as a string in the API format, exports as matchExpressions
+LabelSelector.prototype.exportJSON = function() {
+  var REVERSE_OPERATOR_MAP = {
+    "in": "In",
+    "not in": "NotIn",
+    "exists": "Exists",
+    "does not exist": "DoesNotExist"
+  };
```
Comment (spadgett) [inline labelSelector.js]: Might be better to keep this beside OPERATOR_MAP so it's obvious if you change one you need to change the other.

**Review by spadgett — COMMENTED** [review]
Just a few comments.

**Comment by jwforres** [comment]
@spadgett updated

**Review by spadgett — APPROVED** [review]

**Pull request merged**